source/fundamentals/aggregation-expression-operations.txt

```diff
@@ -979,7 +976,11 @@ the Query API:
 .. code-block:: javascript
    :copyable: true
 
-   [ { $match: { $eq: [ "$currentAddress.state", "WA" ] } } ]
+   [ 
+       { $addFields: { a: { $objectToArray: { $getField: "mailing.address" } } } },
+       { $match: { "a.k": "state", "a.v": "WA" } },
+       { $project: { a: 0 } }
```

Comment:
I think this should only have one stage, because the Java only specifies one stage. I think something like: 

`[ { $match: { $eq: [  { $getField: "mailing.address" }, "WA" ] } } ]`

but it would need to be confirmed.

Reply:
I did a lot of testing for this, and ran into a lot of trouble because the top level field name contains a period. I really couldn't figure out how to write the valid MQL for this using a plain match stage!